src/controllers/pokemon_controller.ts

```diff
@@ -37,8 +38,10 @@ export default class PokemonController {
         const id = res.data.id;
         const types = res.data.types.map((e: PokeType) => e.type.name);
         const sprite = res.data.sprites.front_default
-        return Pokemon.create({ name, url, id, sprite, types })
+        const result = await Pokemon.findOrCreate({ where: { name, url, external_id: id, sprite, types } });
```

Comment:
Although you can do this, you'd most likely want to wrap this in a transaction. What I'd recommend doing is improving the db schema to have external_id: { unique: true } and doing an upsert - very similar to what we did with the sendgrid templates.